src/mds/MDCache.cc

```diff
@@ -4051,9 +4051,12 @@ void MDCache::rejoin_send_rejoins()
     for (map<client_t, set<mds_rank_t> >::iterator p = client_exports.begin();
 	 p != client_exports.end();
 	 ++p) {
-      entity_inst_t inst = mds->sessionmap.get_inst(entity_name_t::CLIENT(p->first.v));
-      for (set<mds_rank_t>::iterator q = p->second.begin(); q != p->second.end(); ++q)
-	rejoins[*q]->client_map[p->first] = inst;
+      Session *session = mds->sessionmap.get_session(entity_name_t::CLIENT(p->first.v));
+      for (set<mds_rank_t>::iterator q = p->second.begin(); q != p->second.end(); ++q) {
```

Comment:
Please no new loops with unnecessary iterators. (Instead use `for (auto &rank : p->second) {`...)